python-avd/pyavd/j2filters/natural_sort.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+import re
+
+from jinja2.runtime import Undefined
+from jinja2.utils import Namespace
+
+
+def convert(text):
```

Comment:
We can define variable and return type:`def convert(text: str) -> int | str:` similar for `natural_sort` as well.

Reply:
Adding this will also require a `from __future__ import annotations` since Python 3.9 will not accept this syntax.